plugins/modules/grafana_datasource.py

```diff
@@ -553,6 +553,9 @@ def compare_datasources(new, current, compareSecureData=True):
         del current["password"]
     if "basicAuthPassword" in current:
         del current["basicAuthPassword"]
+    if current["type"] == "grafana-postgresql-datasource" and new["type"] == "postgres":
```

Comment:
This reads like a potential idempotency-issue that was resolved? If yes, do we need additional tests?

Reply:
Yes, you're right. The problem was, that for grafana > 10.2.3 the `postgres` type changed to `grafana-postgresql-datasource`. Whether you're are creating the datasource with the old or new one, it's stored as `grafana-postgresql-datasource`. For now I didn't change the datasource type for the module, because it remains as an official alias anyway and we don't have to change anything else. :)